File: algorithms/solutions/java/_0049_GroupAnagrams.java
package hot100;

import java.util.*;

public class _0049_GroupAnagrams {
    // @lc code=start
    class Solution {
        public List<List<String>> groupAnagrams(String[] strs) {
            if (strs.length == 0) return Collections.emptyList();
            Map<String, List<String>> anagramsMap = new ArrayList();
            for(String str : strs){
                char[] chars = str.toCharArray();
                Arrays.sort(chars);
                String sorted = new String(chars);
                anagramsMap.computeIfAbsent(sorted, k -> new ArrayList<>()).add(str);
            }
            List<List<String>> result = new ArrayList<>();
            for(Map.Entry<String, List<String>> entry : anagramsMap.entrySet()){
                result.add(entry.getValue());
            }
            return result;
        }
    }
// @lc code=end
}

File: algorithms/solutions/python3/0049_group_anagrams.py

# @lc app=leetcode.cn id=49 lang=python3

# @lc code=start
from typing import List


class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        if not strs:
            return []
        anagrams_map = {}
        for str in strs:
            sorted_str = ''.join(sorted(str))
            anagrams_list = [] if sorted_str not in anagrams_map else anagrams_map[sorted_str]
            anagrams_list.append(str)
            anagrams_map[sorted_str] = anagrams_list
        res = []
        for _, values in anagrams_map.items():
            res.append(values)
        return res
        
# @lc code=end

